crates/bevy_ui/src/widget/image.rs

```diff
@@ -40,26 +41,50 @@ impl Measure for ImageMeasure {
         &self,
         width: Option<f32>,
         height: Option<f32>,
-        _: AvailableSpace,
-        _: AvailableSpace,
+        available_width: AvailableSpace,
+        available_height: AvailableSpace,
+        style: &taffy::Style,
     ) -> Vec2 {
-        let mut size = self.size;
-        match (width, height) {
-            (None, None) => {}
-            (Some(width), None) => {
-                size.y = width * size.y / size.x;
-                size.x = width;
-            }
-            (None, Some(height)) => {
-                size.x = height * size.x / size.y;
-                size.y = height;
-            }
-            (Some(width), Some(height)) => {
-                size.x = width;
-                size.y = height;
-            }
+        // Convert available width/height into an option
+        let parent_width = available_width.into_option();
+        let parent_height = available_height.into_option();
+
+        // Resolve styles
+        let s_aspect_ratio = style.aspect_ratio;
+        let s_width = style.size.width.maybe_resolve(parent_width);
+        let s_min_width = style.min_size.width.maybe_resolve(parent_width);
+        let s_max_width = style.max_size.width.maybe_resolve(parent_width);
+        let s_height = style.size.height.maybe_resolve(parent_height);
+        let s_min_height = style.min_size.height.maybe_resolve(parent_height);
+        let s_max_height = style.max_size.height.maybe_resolve(parent_height);
+
+        // Determine width and height from styles and known_sizes (if a size is available
+        // from any of these sources)
+        let width = width.or(s_width
+            .or(s_min_width)
+            .maybe_clamp(s_min_width, s_max_width));
+        let height = height.or(s_height
+            .or(s_min_height)
+            .maybe_clamp(s_min_height, s_max_height));
+
+        // Use aspect_ratio from style, fall back to inherent aspect ratio
+        let aspect_ratio = s_aspect_ratio.unwrap_or_else(|| self.size.x / self.size.y);
+
+        // Apply aspect ratio
+        // If either width or height was determined at this point, then both are beyond this point.
```

Comment:
```suggestion
        // If only one of width or height was determined at this point, then the other is set beyond this point using the aspect ratio.
```
this feels more explicit